### ISSUE TITLE
TypeError: _parse_lambda()

### ISSUE THREAD

**Issue by hughcameron** [TobikoData/sqlmesh#908]
I'm encountering an error when trying to run sqlmesh plan prod following an upgrade to sqlmesh 0.12.1. Please see below the stack trace.

```bash
sqlmesh plan prod
Traceback (most recent call last):
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/bin/sqlmesh", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/click/core.py", line 1654, in invoke
    super().invoke(ctx)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlmesh/cli/__init__.py", line 20, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlmesh/cli/main.py", line 56, in cli
    context = Context(
              ^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlmesh/core/context.py", line 250, in __init__
    self.load()
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlmesh/core/context.py", line 344, in load
    project = self._loader.load(self)
              ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlmesh/core/loader.py", line 133, in load
    models = self._load_models(macros, hooks, jinja_macros)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlmesh/core/loader.py", line 263, in _load_models
    models = self._load_sql_models(macros, hooks, jinja_macros)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlmesh/core/loader.py", line 305, in _load_sql_models
    model = cache.get_or_load_model(path, _load)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlmesh/core/loader.py", line 397, in get_or_load_model
    model = self._model_cache.get_or_load(
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlmesh/core/model/cache.py", line 50, in get_or_load
    loaded_model = loader()
                   ^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlmesh/core/loader.py", line 286, in _load
    expressions = parse(
                  ^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlmesh/core/dialect.py", line 433, in parse
    for expression in dialect.parser().parse(chunk, sql):
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 817, in parse
    return self._parse(
           ^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 880, in _parse
    expressions.append(parse_method(self))
                       ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlmesh/core/dialect.py", line 80, in _parse_statement
    return self.__parse_statement()
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 1067, in _parse_statement
    expression = self._parse_set_operations(expression) if expression else self._parse_select()
                                                                           ^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 1822, in _parse_select
    cte = self._parse_with()
          ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlmesh/core/dialect.py", line 136, in _parse_with
    return self.__parse_with()
           ^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 1908, in _parse_with
    expressions.append(self._parse_cte())
                       ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 1928, in _parse_cte
    this=self._parse_wrapped(self._parse_statement),
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 4105, in _parse_wrapped
    parse_result = parse_method()
                   ^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlmesh/core/dialect.py", line 80, in _parse_statement
    return self.__parse_statement()
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 1067, in _parse_statement
    expression = self._parse_set_operations(expression) if expression else self._parse_select()
                                                                           ^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 1858, in _parse_select
    expressions = self._parse_csv(self._parse_expression)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 4070, in _parse_csv
    parse_result = parse_method()
                   ^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 2676, in _parse_expression
    return self._parse_alias(self._parse_conjunction())
                             ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 2679, in _parse_conjunction
    return self._parse_tokens(self._parse_equality, self.CONJUNCTION)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 4079, in _parse_tokens
    this = parse_method()
           ^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 2682, in _parse_equality
    return self._parse_tokens(self._parse_comparison, self.EQUALITY)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 4079, in _parse_tokens
    this = parse_method()
           ^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 2685, in _parse_comparison
    return self._parse_tokens(self._parse_range, self.COMPARISON)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 4079, in _parse_tokens
    this = parse_method()
           ^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 2688, in _parse_range
    this = self._parse_bitwise()
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 2780, in _parse_bitwise
    this = self._parse_term()
           ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 2803, in _parse_term
    return self._parse_tokens(self._parse_factor, self.TERM)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 4079, in _parse_tokens
    this = parse_method()
           ^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 2806, in _parse_factor
    return self._parse_tokens(self._parse_unary, self.FACTOR)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 4079, in _parse_tokens
    this = parse_method()
           ^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 2811, in _parse_unary
    return self._parse_at_time_zone(self._parse_type())
                                    ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 2820, in _parse_type
    this = self._parse_column()
           ^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 2969, in _parse_column
    this = self._parse_field()
           ^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 3072, in _parse_field
    or self._parse_function()
       ^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 3104, in _parse_function
    this = parser(self)
           ^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 689, in <lambda>
    "STRUCT": lambda self: self._parse_struct(),
                           ^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 3792, in _parse_struct
    return exp.Struct.from_arg_list(self._parse_csv(lambda: self._parse_lambda(alias=True)))
                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 4065, in _parse_csv
    parse_result = parse_method()
                   ^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/qry/lib/python3.11/site-packages/sqlglot/parser.py", line 3792, in <lambda>
    return exp.Struct.from_arg_list(self._parse_csv(lambda: self._parse_lambda(alias=True)))
                                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: _parse_lambda() got an unexpected keyword argument 'alias'
```

Also, here's a list of versions I'm running:

```bash
❯ pip list | grep sql
sqlglot                       14.0.0
sqlmesh                       0.12.1
sqlparse                      0.4.4
```

**Comment by GeorgeSittas** [comment]
Hey @hughcameron 👋

This has been fixed in current main already, just need to tag a new release. Probably gonna do it a bit later, wanna coordinate with the team first.

cc @tobymao

**Comment by GeorgeSittas** [comment]
The fix has been deployed in v0.12.2: https://github.com/TobikoData/sqlmesh/releases/tag/v0.12.2

**Comment by hughcameron** [comment]
Thanks @GeorgeSittas - can confirm successful on my side as well.
